datalad/distribution/uninstall.py

```diff
@@ -156,7 +156,7 @@ def __call__(
             # check that we have no top-level datasets and not files to process
             if ap.get('type') == 'dataset' and \
                     not ap.get('state', None) == 'absent' and \
-                    path_is_under(ap['path']):
+                    path_is_under([ap['path']]):  # wants a sequence!
```

Comment:
yikes... could it be made to work correctly with both?  use `assure_list` inside?